File: frontend/src/api/schemas/dtos/schema-draft-change-dto.ts
/*
SPDX-License-Identifier: MIT
Copyright (c) 2025 Pablo Ulloa Santin
*/

export type SchemaDraftChangeDto = {
  path: string;
  baseValue: unknown;
  basePresent: boolean;
  draftValue: unknown;
  draftPresent: boolean;
  currentValue: unknown;
  currentPresent: boolean;
  draftChanged: boolean;
  conflict: boolean;
};

File: api/src/main/java/dev/ulloasp/mlsuite/schema/application/dto/SchemaDraftChangeDto.java
package dev.ulloasp.mlsuite.schema.application.dto;

public record SchemaDraftChangeDto(
        String path,
        Object baseValue,
        Object draftValue,
        Object currentValue,
        boolean basePresent,
        boolean draftPresent,
        boolean currentPresent,
        boolean draftChanged,
        boolean conflict) {
}
